Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 1319, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Make observation distributions
emission_dim = 2
num_lags = 1

keys = jr.split(jr.PRNGKey(0), num_states)
angles = jnp.linspace(0, 2 * jnp.pi, num_states, endpoint=False)
theta = jnp.pi / 25 # rotational frequency
weights = jnp.array([0.8 * random_rotation(key, emission_dim, theta=theta) for key in keys])
biases = jnp.column_stack([jnp.cos(angles), jnp.sin(angles), jnp.zeros((num_states, emission_dim - 2))])
covariances = jnp.tile(0.001 * jnp.eye(emission_dim), (num_states, 1, 1))

# Compute the stationary points
stationary_points = jnp.linalg.solve(jnp.eye(emission_dim) - weights, biases)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mValueError[0m                                Traceback (most recent call last)
Cell [0;32mIn[5], line 13[0m
[1;32m     10[0m covariances [38;5;241m=[39m jnp[38;5;241m.[39mtile([38;5;241m0.001[39m [38;5;241m*[39m jnp[38;5;241m.[39meye(emission_dim), (num_states, [38;5;241m1[39m, [38;5;241m1[39m))
[1;32m     12[0m [38;5;66;03m# Compute the stationary points[39;00m
[0;32m---> 13[0m stationary_points [38;5;241m=[39m [43mjnp[49m[38;5;241;43m.[39;49m[43mlinalg[49m[38;5;241;43m.[39;49m[43msolve[49m[43m([49m[43mjnp[49m[38;5;241;43m.[39;49m[43meye[49m[43m([49m[43memission_dim[49m[43m)[49m[43m [49m[38;5;241;43m-[39;49m[43m [49m[43mweights[49m[43m,[49m[43m [49m[43mbiases[49m[43m)[49m

    [0;31m[... skipping hidden 13 frame][0m

File [0;32m/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/jax/_src/numpy/linalg.py:1345[0m, in [0;36msolve[0;34m(a, b)[0m
[1;32m   1342[0m [38;5;66;03m# Check for invalid inputs that previously would have led to a batched 1D solve:[39;00m
[1;32m   1343[0m [38;5;28;01mif[39;00m (b[38;5;241m.[39mndim [38;5;241m>[39m [38;5;241m1[39m [38;5;129;01mand[39;00m a[38;5;241m.[39mndim [38;5;241m==[39m b[38;5;241m.[39mndim [38;5;241m+[39m [38;5;241m1[39m [38;5;129;01mand[39;00m
[1;32m   1344[0m     a[38;5;241m.[39mshape[[38;5;241m-[39m[38;5;241m1[39m] [38;5;241m==[39m b[38;5;241m.[39mshape[[38;5;241m-[39m[38;5;241m1[39m] [38;5;129;01mand[39;00m a[38;5;241m.[39mshape[[38;5;241m-[39m[38;5;241m1[39m] [38;5;241m!=[39m b[38;5;241m.[39mshape[[38;5;241m-[39m[38;5;241m2[39m]):
[0;32m-> 1345[0m   [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m(
[1;32m   1346[0m     [38;5;124mf[39m[38;5;124m"[39m[38;5;124mInvalid shapes for solve: [39m[38;5;132;01m{[39;00ma[38;5;241m.[39mshape[38;5;132;01m}[39;00m[38;5;124m, [39m[38;5;132;01m{[39;00mb[38;5;241m.[39mshape[38;5;132;01m}[39;00m[38;5;124m. Prior to JAX v0.5.0,[39m[38;5;124m"[39m
[1;32m   1347[0m     [38;5;124m"[39m[38;5;124m this would have been treated as a batched 1-dimensional solve.[39m[38;5;124m"[39m
[1;32m   1348[0m     [38;5;124m"[39m[38;5;124m To recover this behavior, use solve(a, b[..., None]).squeeze(-1).[39m[38;5;124m"[39m)
[1;32m   1350[0m signature [38;5;241m=[39m [38;5;124m"[39m[38;5;124m(m,m),(m)->(m)[39m[38;5;124m"[39m [38;5;28;01mif[39;00m b[38;5;241m.[39mndim [38;5;241m==[39m [38;5;241m1[39m [38;5;28;01melse[39;00m [38;5;124m"[39m[38;5;124m(m,m),(m,n)->(m,n)[39m[38;5;124m"[39m
[1;32m   1351[0m [38;5;28;01mreturn[39;00m jnp[38;5;241m.[39mvectorize(lax_linalg[38;5;241m.[39m_solve, signature[38;5;241m=[39msignature)(a, b)

[0;31mValueError[0m: Invalid shapes for solve: (5, 2, 2), (5, 2). Prior to JAX v0.5.0, this would have been treated as a batched 1-dimensional solve. To recover this behavior, use solve(a, b[..., None]).squeeze(-1).

